shared-ts/jsonTypes.ts

```diff
@@ -0,0 +1,263 @@
+    march,
+    procession,
+    demonstration,
+    publicDisturbance,
+    filterBlockade,
+    sightseersObstructingAccess,
+    bridgeStrike,
+    overheadObstruction,
+    undefinedProblem,
+    problemsAtBorderPost,
+    problemsAtCustomsPost,
+    problemsOnLocalRoad
+}
+
+enum PersonnelReason {
+    unknown,
+    staffSickness,
+    staffInjury,
+    contractorStaffInjury,
+    staffAbsence,
+    staffInWrongPlace,
+    staffShortage,
+    industrialAction,
+    unofficialIndustrialAction,
+    workToRule,
+    undefinedPersonnelProblem
+}
+
+enum EquipmentReason {
+    unknown,
+    pointsProblem,
+    pointsFailure,
+    signalProblem,
+    trainWarningSystemProblem,
+    trackCircuitProblem,
+    signalFailure,
+    derailment,
+    engineFailure,
+    tractionFailure,
+    breakDown,
+    technicalProblem,
+    brokenRail,
+    poorRailConditions,
+    wheelImpactLoad,
+    lackOfOperationalStock,
+    defectiveFireAlarmEquipment,
+    defectivePlatformEdgeDoors,
+    defectiveCctv,
+    defectivePublicAnnouncementSystem,
+    ticketingSystemNotAvailable,
+    repairWork,
+    constructionWork,
+    maintenanceWork,
+    emergencyEngineeringWork,
+    lateFinishToEngineeringWork,
+    powerProblem,
+    fuelProblem,
+    swingBridgeFailure,
+    escalatorFailure,
+    liftFailure,
+    gangwayProblem,
+    closedForMaintenance,
+    fuelShortage,
+    deicingWork,
+    wheelProblem,
+    luggageCarouselProblem,
+    undefinedEquipmentProblem
+}
+
+enum EnvironmentReason {
+    unknown,
+    fog,
+    roughSea,
+    heavySnowFall,
+    driftingSnow,
+    blizzardConditions,
+    heavyRain,
+    strongWinds,
+    stormConditions,
+    stormDamage,
+    tidalRestrictions,
+    highTide,
+    lowTide,
+    ice,
+    frozen,
+    hail,
+    sleet,
+    highTemperatures,
+    flooding,
+    waterlogged,
+    lowWaterLevel,
+    highWaterLevel,
+    fallenLeaves,
+    fallenTree,
+    landslide,
+    undefinedEnvironmentalProblem,
+    lightningStrike,
+    sewerOverflow,
+    grassFire
+}
+
+enum DayType {
+    monday,
+    tuesday,
+    wednesday,
+    thursday,
+    friday,
+    saturday,
+    sunday
+}
+
+export type Source = {
+    SourceType: SourceType;
+    TimeOfCommunication: string;
+};
+
+export type Period = {
+    StartTime: Date;
+    EndTime?: Date;
+};
+
+export type InfoLink = {
+    Uri: string;
+}
+
+
+export type Reference = {
+    RelatedToRef: string[];
+};
+
+export type Repetitions = {
+    DayType: DayType[];
+}
+
+export type PtSituationElement = {
+    CreationTime: Date;
+    ParticipantRef: string;
+    SituationNumber: string;
+    Version?: Number;
+    References?: Reference;
+    Source: Source;
+    Progress: Progress;
+    ValidityPeriod: Period[];
+    Repetitions?: Repetitions;
+    PublicationWindow: Period;
+    ReasonType?: MiscellaneousReason | PersonnelReason | EquipmentReason | EnvironmentReason;
+    Planned: boolean;
+    Summary: string;
+    Description: string;
+    InfoLinks?: InfoLink[];
```

Comment:
This would result in multiple InfoLinks items in the XML when we want a single InfoLinks with multiple InfoLink items below it so we may just need an InfoLinks type:
```
type InfoLinks = {
  InfoLink: InfoLink[];
}
```
then this can be used at the top level `InfoLinks?: InfoLinks;`